android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,629 @@
+				} else {
+					// photo
+					ImageCapture.Builder imageCaptureBuilder = new ImageCapture.Builder()
+						.setFlashMode(cameraFlashMode)
+						.setTargetRotation(rotation);
+
+					if (aspectRatio != -1) {
+						imageCaptureBuilder.setTargetAspectRatio(aspectRatio);
+					}
+
+					if (targetResolutionWidth != -1 && targetResolutionHeight != -1) {
+						imageCaptureBuilder.setTargetResolution(
+							new Size(targetResolutionWidth, targetResolutionHeight));
+					}
+
+					imageCapture = imageCaptureBuilder.build();
+					camera = cameraProvider.bindToLifecycle(this, cameraSelector, imageCapture, preview);
+				}
+				preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
+				if (allowZoom) {
+					// zoom gesture
+					ScaleGestureDetector scaleGestureDetector = new ScaleGestureDetector(activity,
+						new ScaleGestureDetector.SimpleOnScaleGestureListener()
+						{
+							@Override
+							public boolean onScale(ScaleGestureDetector detector)
+							{
+								float scale = camera.getCameraInfo().getZoomState()
+									.getValue().getZoomRatio() * detector.getScaleFactor();
+								camera.getCameraControl().setZoomRatio(scale);
+								return true;
+							}
+						});
+
+					viewFinder.setOnTouchListener((v, event) -> {
+						scaleGestureDetector.onTouchEvent(event);
+						v.performClick();
+						return true;
+					});
+				}
+				if (openCallback != null) {
+					KrollDict response = new KrollDict();
+					response.put("width", viewFinder.getWidth());
+					response.put("height", viewFinder.getHeight());
+					response.put("top", viewFinder.getTop());
+					response.put("right", viewFinder.getRight());
+					response.put("left", viewFinder.getLeft());
+					response.put("bottom", viewFinder.getBottom());
+					openCallback.callAsync(callbackContext, response);
+				}
+			} catch (InterruptedException | ExecutionException e) {
+				// Currently no exceptions thrown. cameraProviderFuture.get()
+				// shouldn't block since the listener is being called, so no need to
+				// handle InterruptedException.
+			}
+		}, ContextCompat.getMainExecutor(activity));
+	}
+
+	@Override
+	protected void onResume()
+	{
+		super.onResume();
+		cameraActivity = this;
+		if (localOverlayProxy != null) {
+			View overlayView = localOverlayProxy.getOrCreateView().getNativeView();
+			ViewGroup parent = (ViewGroup) overlayView.getParent();
+			if (parent != null) {
+				parent.removeView(overlayView);
+			}
+			layout.addView(overlayView, new FrameLayout.LayoutParams(FrameLayout.LayoutParams.MATCH_PARENT,
+				FrameLayout.LayoutParams.MATCH_PARENT));
+		}
+	}
+
+	@Override
+	protected void onDestroy()
+	{
+		// Release our camera activity reference.
+		if (cameraActivity == this) {
+			cameraActivity = null;
+		}
+		if (recording != null) {
+			recording.close();
+			recording = null;
+		}
+		if (pendingRecording != null) {
+			pendingRecording = null;
+		}
+		if (camera != null) {
+			camera = null;
+		}
```

Comment:
Any more strong references, e.g. `callbackContext` and `overlayProxy`? It would be good to run the profiler for any memory leaks.

Reply:
Good point. I'll check that. The current TiCameraActivity doesn't do more so if that works fine it will be  the same in TiCameraXActivity.